xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-default/src/main/java/org/xwiki/refactoring/internal/DefaultMacroRefactoring.java

```diff
@@ -0,0 +1,143 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.refactoring.internal;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.component.manager.ComponentManager;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.refactoring.ReferenceRenamer;
+import org.xwiki.rendering.block.Block;
+import org.xwiki.rendering.block.MacroBlock;
+import org.xwiki.rendering.block.XDOM;
+import org.xwiki.rendering.macro.Macro;
+import org.xwiki.rendering.macro.MacroContentParser;
+import org.xwiki.rendering.macro.MacroExecutionException;
+import org.xwiki.rendering.macro.MacroId;
+import org.xwiki.rendering.macro.MacroLookupException;
+import org.xwiki.rendering.macro.MacroManager;
+import org.xwiki.rendering.macro.MacroRefactoring;
+import org.xwiki.rendering.macro.MacroRefactoringException;
+import org.xwiki.rendering.macro.descriptor.ContentDescriptor;
+import org.xwiki.rendering.renderer.BlockRenderer;
+import org.xwiki.rendering.renderer.printer.DefaultWikiPrinter;
+import org.xwiki.rendering.syntax.Syntax;
+import org.xwiki.rendering.transformation.MacroTransformationContext;
+import org.xwiki.rendering.transformation.RenderingContext;
+
+/**
+ * Default implementation of {@link MacroRefactoring}.
+ * This implementation tries to parse the macro content if it's of type
+ * {@link org.xwiki.rendering.block.Block#LIST_BLOCK_TYPE} and replace the link and image references.
+ *
+ * @version $Id$
+ * @since 13.4RC1
+ */
+@Component
+@Singleton
+public class DefaultMacroRefactoring implements MacroRefactoring
+{
+    @Inject
+    private MacroManager macroManager;
+
+    @Inject
+    private MacroContentParser macroContentParser;
+
+    @Inject
+    private Provider<ReferenceRenamer> referenceRenamerProvider;
+
+    @Inject
+    @Named("context")
+    private ComponentManager componentManager;
+
+    @Inject
+    private RenderingContext renderingContext;
+
+    private boolean shouldMacroContentBeParsed(Macro<?> macro)
+    {
+        ContentDescriptor contentDescriptor = macro.getDescriptor().getContentDescriptor();
+        return Block.LIST_BLOCK_TYPE.equals(contentDescriptor.getType());
+    }
+
+    private Macro<?> getMacro(MacroBlock macroBlock) throws MacroRefactoringException
+    {
+        MacroId macroId = new MacroId(macroBlock.getId(), this.renderingContext.getDefaultSyntax());
```

Comment:
This is not fully accurate, you should first look at `macroBlock`'s parents for the syntax before fallbacking on the context syntax.